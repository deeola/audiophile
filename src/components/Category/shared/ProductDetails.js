import React from 'react'
import { NewProduct, ProductContainer, SubTitle, Title } from './SharedStyling'

const ProductDetails = ({title, subtitle, np }) => {
  return (
    <ProductContainer>
        <NewProduct data-testid='np' > {np}</NewProduct>
        <Title data-testid='title' >{title}</Title>
        <SubTitle data-testid='subtitle' >{subtitle}</SubTitle>
    </ProductContainer>
  )
}

export default ProductDetails

