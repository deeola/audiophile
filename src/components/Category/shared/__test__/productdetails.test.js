import { render} from "@testing-library/react";
import reactDom from "react-dom";
import renderer from "react-test-renderer";
import ProductDetails from "../ProductDetails";


it("renders div correctly", () => {
  const div = document.createElement("div");

  reactDom.render(
    <ProductDetails />,

    div
  );
});

it("renders texts correctly", () => {
  const { getByTestId } = render(<ProductDetails subtitle={'SUBTITLE'} title={'TITLE'} np='NEW PRODUCT' />);

  expect(getByTestId("np")).toHaveTextContent("NEW PRODUCT");
  expect(getByTestId("title")).toHaveTextContent("TITLE");
  expect(getByTestId("subtitle")).toHaveTextContent("SUBTITLE");
});


//create snapshot

it('matches snapshot', () => {
const tree = renderer.create(
    <ProductDetails />
).toJSON();

expect(tree).toMatchSnapshot();
})