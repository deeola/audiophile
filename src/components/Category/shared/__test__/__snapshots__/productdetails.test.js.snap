// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="sc-eCYdqJ bMkDvt"
>
  <p
    className="sc-iBkjds dIztKO"
    data-testid="np"
  >
     
  </p>
  <h2
    className="sc-jSMfEi imCpJA"
    data-testid="title"
  />
  <p
    className="sc-gKXOVf dgecyH"
    data-testid="subtitle"
  />
</div>
`;
